samples/bluetooth/unicast_audio_client/src/main.c

```diff
@@ -141,23 +157,52 @@ static void lc3_audio_timer_timeout(struct k_work *work)
 			net_buf_unref(buf);
 		} else {
 			printk("  TX LC3 l: %zu\n", tx_sdu_len);
-			pdu_cnt++;
+			sdu_cnt++;
 		}
 	}
 }
 
 static void init_lc3(void)
 {
-	/* Fill audio buffer with Sine wave */
-	fill_audio_buf_sin(audio_buf, AUDIO_LENGTH_US, AUDIO_FREQUENCY_HZ,
-			   AUDIO_SAMPLE_RATE_HZ);
+	unsigned int num_samples;
+
+	freq_hz = bt_codec_cfg_get_freq(&codec_configuration.codec);
+	frame_duration_us = bt_codec_cfg_get_frame_duration_us(&codec_configuration.codec);
+	octets_per_frame = bt_codec_cfg_get_octets_per_frame(&codec_configuration.codec);
+	frames_per_sdu = bt_codec_cfg_get_frames_per_sdu(audio_stream.codec, true);
+	octets_per_frame = bt_codec_cfg_get_octets_per_frame(audio_stream.codec);
+
+	if (freq_hz < 0) {
+		printk("Error: Codec frequency not set, cannot start codec.");
+		return;
+	}
+
+	if (frame_duration_us < 0) {
+		printk("Error: Frame duration not set, cannot start codec.");
+		return;
+	}
+
+	if (octets_per_frame < 0) {
+		printk("Error: Octets per frame not set, cannot start codec.");
+		return;
+	}
+
+	frame_duration_100us = frame_duration_us / 100;
+
+
+	/* Fill audio buffer with Sine wave only once and repeat encoding the same tone frame */
+	fill_audio_buf_sin(audio_buf, frame_duration_us, AUDIO_TONE_FREQUENCY_HZ, freq_hz);
```

Comment:
I wonder if it would make sense to create a function in Zephyr to generate such test data and guard with e.g. BT_TESTING.

Reply:
Maybe, but not sure where that would be. 
Also to get a proper sine in cases where the sample frequency is not a multiple of the tone-frequency you cannot use it as is done here, but would need to generate a continuation based on the ending point of the last buffer generated.
I would prefer to have it here.